File: python-2.5.2/TestEnv.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
# vi:encoding=utf-8

class TestEnv:

 def __init__( self):

  self. countOk= 0
  self. countFail= 0
  self. countLatest= 0
  self. countRelativeExpected= 0


 def test ( self, b):
  
  if( b):
   
   self. countOk += 1

  else:
   
   self. countFail += 1

   print "test failed"
   import traceback
   traceback.print_stack()
 
 def checkComplainAndAdjustExpected( self, count):

  countTests= self. countOk + self. countFail
  countTestsLocal= countTests - self. countLatest

  self. countLatest= countTests

  if count == countTestsLocal:
   
   return True
  
  else:

   print "Expected Test Count (from last checkpoint to here) doesn't match: countTestsLocal:" + str( countTestsLocal) + " count: "+ str( count)
   import traceback
   traceback.print_stack()

   return False

 def result( self):
  
  return "tests: ok: "+ str( self. countOk)+ " fail: "+ str( self. countFail)+ " countLatest: "+ str( self. countLatest)

File: c/lambda-calculus/testenv.c

#include <stdio.h>
#include <stdarg.h>
#include <stdlib.h>

#include "testenv.h"

static int countOk= 0;
static int countFail= 0;
static int countLatest= 0;

void init()
{
 countOk= 0;
 countFail= 0;
 countLatest= 0;
}

void test( int b)
{
 if( b)
 {
  countOk+= 1;
 }
 else
 {
  countFail+= 1;
 }
}

int checkComplainAndAdjustExpected( int count) 
{ 
 int countTests= countOk + countFail; 
 int countTestsLocal= countTests - countLatest; 

 countLatest= countTests; 

 if( count == countTestsLocal) 
 { 
  return 1; 
 } 
 else 
 { 
  printf( "Expected Test Count (from last checkpoint to here) doesn't match: countTestsLocal:%d count: %d\n", countTestsLocal, count);
  return 0; 
 } 
} 

static char result_ret[1024]= "\n";

char* result()
{
 char* ret= result_ret;
 sprintf( result_ret, "tests: ok: %d fail: %d countLatest: %d", countOk, countFail, countLatest);
 return ret;
}



